src/products/utils/handlers.ts

```diff
@@ -32,6 +34,34 @@ export function createChannelsPriceChangeHandler(
   };
 }
 
+export function createChannelsPreorderChangeHandler(
+  channelListings: ChannelData[],
+  updateChannels: (data: ChannelData[]) => void,
+  triggerChange: () => void
+) {
+  return (id: string, preorderData: ChannelPreorderArgs) => {
+    const { preorderThreshold, unitsSold } = preorderData;
+
+    const channelIndex = channelListings.findIndex(
+      channel => channel.id === id
+    );
+    const channel = channelListings[channelIndex];
+
+    const updatedChannels = [
+      ...channelListings.slice(0, channelIndex),
+      {
+        ...channel,
+        preorderThreshold,
+        unitsSold
+      },
+      ...channelListings.slice(channelIndex + 1)
+    ];
```

Comment:
This block could be replaced with `.map` function:
```
const updatesChannels = channelListings.map(channel =>
  channel.id === id
    ? { ...channel, preorderThreshold, unitsSold }
    : channel
);
```

Reply:
Much more readable indeed.